networkx/algorithms/tree/mst.py

```diff
@@ -1190,19 +1190,19 @@ def number_of_spanning_trees(G, *, root=None, weight=None):
             Or the sum of all spanning tree weights of the graph `G`
             where the weight of a tree is the product of its edge weights.
         Directed graphs:
-            The number of arborescence of `G` rooted at node `root`.
+            The number of arborescences of `G` rooted at node `root`.
             Or the sum of all arborescence weights of the graph `G` with
-            specified root where the weight of a tree is the product of
-            its edge weights.
+            specified root where the weight of an arborescence is the product
+            of its edge weights.
 
     Raises
     ------
     NetworkXPointlessConcept
         If `G` does not contain any nodes.
 
     NetworkXError
-        If the graph `G` is not (weakly) connected,
-        or if `G` is directed and the root node is not specified or not in G.
+        If the graph `G` is directed and the root node
+        is not specified or is not in G.
```

Comment:
No. Based on your suggestion, I've changed it to return 0. That also makes more sense, because the number of spanning trees of a disconnected graph has a perfectly valid answer: 0